packages/desktop/src/renderer/app/stores/reducer.ts

```diff
@@ -1131,6 +1219,165 @@ export const environmentReducer = (
+              if (afterUUID) {
+                afterIndex = environment.callbacks.findIndex(
+                  (cb) => cb.uuid === afterUUID
+                );
+                if (afterIndex === -1) {
+                  afterIndex = callbacks.length;
+                }
+              }
+              callbacks.splice(afterIndex + 1, 0, newCallback);
+
+              return {
+                ...environment,
+                callbacks
+              };
+            }
+
+            return environment;
+          }),
+          filters: {
+            ...state.filters,
+            callbacks: ''
+          },
+          environmentsStatus: markEnvStatusRestart(state)
+        };
+        break;
+      }
+
+      newState = state;
+      break;
+    }
+
+    case ActionTypes.REMOVE_CALLBACK: {
+      const activeEnvironment = state.environments.find(
+        (environment) => environment.uuid === state.activeEnvironmentUUID
+      );
+      const deletedCallback = activeEnvironment.callbacks.find(
+        (cb) => cb.uuid === action.callbackUUID
+      );
+
+      const newCallbacks = activeEnvironment.callbacks.filter(
+        (cb) => cb.uuid !== action.callbackUUID
+      );
+
+      const newRoutes = activeEnvironment.routes.map((route) => {
+        let hasChanged = false;
+        const newReponses = route.responses.map((response) => {
+          if (response.callbacks) {
+            const filteredCallbacks = response.callbacks.filter(
+              (cb) => cb.uuid !== deletedCallback.uuid
+            );
+            hasChanged =
+              hasChanged ||
+              filteredCallbacks.length !== response.callbacks.length;
+
+            return { ...response, callbacks: filteredCallbacks };
+          }
+
+          return response;
+        });
+        if (hasChanged) {
+          return { ...route, responses: newReponses };
+        }
+
+        return route;
+      });
+
+      const newEnvironments = state.environments.map((environment) => {
+        if (environment.uuid === state.activeEnvironmentUUID) {
+          return {
+            ...environment,
+            callbacks: newCallbacks,
+            routes: newRoutes
+          };
+        }
+
+        return environment;
+      });
+
+      newState = {
+        ...state,
+        environments: newEnvironments,
+        environmentsStatus: markEnvStatusRestart(state)
+      };
+
+      if (state.activeCallbackUUID === action.callbackUUID) {
+        if (newCallbacks.length) {
+          newState.activeCallbackUUID = newCallbacks[0].uuid;
+        } else {
+          newState.activeCallbackUUID = null;
+        }
+      }
+      break;
+    }
+
+    case ActionTypes.UPDATE_CALLBACK: {
+      const propertiesNeedingRestart: (keyof Callback)[] = [
+        'name',
+        'body',
+        'bodyType',
+        'databucketID',
+        'filePath',
+        'headers',
+        'method',
+        'sendFileAsBody',
+        'uri'
+      ];
```

Comment:
You may have to test a bit, but I think there is no need to mark the environment as "restart needed" as they get directly accessed at the time the call is done. 
The reason why the data buckets mandata a server restart is because they are generated at startup time.

Reply:
Yes, you are correct. I removed this after a round of tests.